interfaces/builtin/hugepages_control.go

```diff
@@ -0,0 +1,49 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const hugepagesControlSummary = `allows controlling hugepages`
+
+const hugepagesControlBaseDeclarationSlots = `
+  hugepages-control:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
+const hugepagesControlConnectedPlugAppArmor = `
+# Description: Allow control to hugepages.
+/sys/kernel/mm/hugepages/{,**} rw,
```

Comment:
should this perhaps be

```suggestion
/sys/kernel/mm/{hugepages,transparent_hugepage}/{,**} rw,
```

For example see some denials from the docker snap where it seems like docker is now trying to look at hugepages: https://forum.snapcraft.io/t/docker-18-09-9-snap-crashes-with-apparmor-denial-on-sys-kernel-mm-transparent-hugepage-hpage-pmd-size/14218.

Also from my basic reading of https://www.kernel.org/doc/Documentation/vm/transhuge.txt it sounds like transparent_hugepage is a better (for some definition of better) hugepage implementation so it may be used more often nowadays?

Reply:
I do not see an issue with adding transparent_hugepage, this interface should be able to be used with multiple snaps that request/require hugepages, like docker or kvm as well.

Reply:
added